src/connection.js

```diff
@@ -121,8 +121,7 @@ export class StableWSConnection {
 
 		this._removeConnectionListeners();
 
-		// reset the wsID;
-		this.wsID = 1;
+		this.wsID += 1;
```

Comment:
This is a disconnect function, which one uses when he wants to manually disconnect the websocket connection e.g. when mobile app goes to background.

For some weird reason, ws hits the original [onclose](https://github.com/GetStream/stream-chat-js/pull/265/files#diff-875da84119066ec244742e95d1209f0cL357) callback (with code `1001`) after `ws.close()` (line 159), which should not happen ideally since we override onclose callback on line 140. This is **NOT** a reproducible issue on our side. But some customer is facing this issue. The logs that he provided didn't help figure out why is this happening.

The main problem is - when original onclose callback gets called, it triggers the reconnection logic ... which defeats the whole purpose manually disconnecting the client. The moment user calls `disconnect`, our code immediately triggers reconnect.

To avoid this, instead of reseting the this.wsID to 1, we can increment it, so that original callback function [will not respond anymore](https://github.com/GetStream/stream-chat-js/pull/265/files#diff-875da84119066ec244742e95d1209f0cL364)

Reply:
hmm, i think you should also move it up, it should be the first thing in the function in my opinion